### PR TITLE
Add Discord Communities

### PR DESCRIPTION
**Pull request by mhxion**
1. PR Reviewed: [__1__](https://github.com/sindresorhus/awesome/pull/1812#pullrequestreview-438679163), [__2__](https://github.com/sindresorhus/awesome/pull/1808#pullrequestreview-438734848), [__3__](https://github.com/sindresorhus/awesome/pull/1811#pullrequestreview-438736118), [__4__](https://github.com/sindresorhus/awesome/pull/1804#pullrequestreview-438821530)
2. The list over 30-days old, has over 150 stars
3. I [__explained here__](https://github.com/mhxion/awesome-discord-communities/blob/main/FAQ.md#why-does-not-this-list-follow-convential-awesome-list-framework) why I'd used `Notable Channels` and [badges](https://github.com/mhxion/awesome-discord-communities/blob/main/badges.md) instead a description for each entry.
4. Only [table-pipe-alignment](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-table-pipe-alignment) markdown lint rule was ignored because as the rule has mentioned, it was impossible to adapt the cell with variable number of images without making the code too messy.
5. I added this under _Gaming_ due to Discord being [more known for that](https://www.howtogeek.com/660601/is-discord-right-for-your-business/).

**https://github.com/mhxion/awesome-discord-communities#readme**

**Over the years hundreds of developer communities have made [Discord](discord.com/) their home for helping newcomers in tech, collaborating with open-source project users, and actively contributing by creating occasional programming challenges; this list aggregates the best of them. While existing helper-platforms like Stack Overflow, forums do their job well, they cannot be used for interacting live (debugging), Discord communities fill this major gap. The developer-friendlier the communities, the more productive collaboration is. As of now this list holds over 200 of them (and growing), each of them had gone through proper [review policy](https://github.com/mhxion/awesome-discord-communities/blob/main/CONTRIBUTING.md#review-policy), and other safety requirements, each carrying useful [links](https://github.com/mhxion/awesome-discord-communities/blob/main/badges.md); no other awesome Discord list exists to this extend until now. By nature this list is correlated with [awesome-slack](https://github.com/filipelinhares/awesome-slack#readme).**

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions.
- You have read and understood the [instructions for creating a list](create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=master&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name or text to the readme. GitHub already shows the license name at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

### CODE REVIEW

**Comment by mhxion** [comment]
unicorn

**Comment by sindresorhus** [comment]
Since this is about Discord communities and not anything Discord, I think the list name should be more explicit. `Awesome Discord Communities`.

**Comment by mhxion** [comment]
@sindresorhus Done [`318866e`](https://github.com/mhxion/awesome-programming-discord/commit/3442530bc907f5ad83a8c857777f419f2ea4ac1a).

**Comment by sindresorhus** [comment]
The PR title and repo title are not in sync with the new readme heading.

**Comment by mhxion** [comment]
Oops. Done @sindresorhus

**Comment by sindresorhus** [comment]
It's not. Please look over everything and make sure it all uses the same URL and naming.

**Comment by mhxion** [comment]
@sindresorhus Done. Should've known GitHub redirects to the old URL after changing it.

**Comment by mhxion** [comment]
@sindresorhus Done. Should've known GitHub redirects to the old URL after changing it.

**Comment by morenoh149** [comment]
Lots of pictures on this one.

File: readme.md [hunk 1]
```diff
@@ -396,6 +396,7 @@
 - [PlayCanvas](https://github.com/playcanvas/awesome-playcanvas#readme) - Game engine.
 - [Game Remakes](https://github.com/radek-sprta/awesome-game-remakes#readme) - Actively maintained open-source game remakes.
 - [Flame](https://github.com/flame-engine/awesome-flame#readme) - Game engine for Flutter.
+- [Discord](https://github.com/mhxion/awesome-discord-communities#readme) - Chat with friends and communities.
```
Comment (sindresorhus) [inline readme.md:399]: The title here is still not correct. Please ensure everything is in sync.

Comment (mhxion) [reply readme.md]: I wasn't sure at first whether I should've added the `Communities` suffix. Thanks, done.

**Comment by mhxion** [comment]
@sindresorhus Hello, wanted to let you know all the previously mentioned issues (mostly with the title) had been [solved](https://github.com/sindresorhus/awesome/pull/1813#discussion_r456651664), no more subtle errors in my knowledge. Maybe I should've mentioned you sooner. Let me know if there's still something to be fixed.

**Comment by sindresorhus** [comment]
> Includes a succinct description of the project/theme at the top of the readme. (Example)

This should include a succinct description of what Discord is.

**Comment by mhxion** [comment]
@sindresorhus Agreed. Updated [`cf7506`](https://github.com/mhxion/awesome-discord-communities/commit/cf750630c3a42eca93eab415fc3a73d53dbdd6ed) with a `--oneline` description:
> [Discord](FAQ.md#what-is-discordcom-exactly) at heart, is a community-oriented instant messaging, voice and video call application.

Everything else is good as before.

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1301120096740483072

**Comment by mhxion** [comment]
Thanks @sindresorhus

**Pull request merged**